src/stim/mem/simd_bits.test.cc

```diff
@@ -274,32 +281,36 @@ TEST_EACH_WORD_SIZE_W(simd_bits, right_shift_assignment, {
             ASSERT_EQ(pattern, 0ULL);
         }
     }
-    m0.clear();
-    simd_bits<W> m1(512), m2(512);
-    m1.randomize(512, SHARED_TEST_RNG());
-    m2 = m1;
-    m1 >>= 1;
-    for (size_t k = 0; k < 511; k++) {
-        ASSERT_EQ(m1[k], m2[k + 1]);
-    }
-    m1.randomize(512, SHARED_TEST_RNG());
-    m2 = m1;
-    m1 >>= 10;
-    for (size_t k = 0; k < 501; k++) {
-        ASSERT_EQ(m1[k], m2[k + 10]);
-    }
-    m1 >>= 0;
-    for (size_t k = 0; k < 501; k++) {
-        ASSERT_EQ(m1[k], m2[k + 10]);
+})
+
+TEST_EACH_WORD_SIZE_W(simd_bits, fuzz_right_shift_assignment, {
+    auto rng = SHARED_TEST_RNG();
+    for (int i = 0; i < 5; i++) {
+        std::uniform_int_distribution dist_bits(1, 1200);
+        int num_bits = dist_bits(rng);
+        simd_bits<W> m1(num_bits), m2(num_bits);
+        m1.randomize(num_bits, rng);
+        m2 = m1;
+        std::uniform_int_distribution dist_shift(0, (int)m1.num_bits_padded());
+        size_t shift = dist_shift(rng);
+        m1 >>= shift;
+        for (size_t k = 0; k < m1.num_bits_padded() - shift; k++) {
+            ASSERT_EQ(m1[k], m2[k + shift]);
```

Comment:
I'd also check the shifted in bits. That said I basically trust the method at this point. Was just worried about the corner cases that come up.

Reply:
Do you mean in m1 check that the bits from num_bits_padded() - shift to num_bits_padded() are zero?

Reply:
Yeah check every single output bit.